src/common/src/array/arrow/arrow_impl.rs

```diff
@@ -936,34 +1059,80 @@ impl FromIntoArrow for Time {
     }
 }
 
-impl FromIntoArrow for Timestamp {
+impl FromIntoArrowWithUnit for Timestamp {
     type ArrowType = i64;
+    type TimestampType = TimeUnit;
 
-    fn from_arrow(value: Self::ArrowType) -> Self {
-        Timestamp(
-            DateTime::from_timestamp((value / 1_000_000) as _, (value % 1_000_000 * 1000) as _)
+    fn from_arrow_with_unit(value: Self::ArrowType, time_unit: Self::TimestampType) -> Self {
+        match time_unit {
+            TimeUnit::Second => {
+                Timestamp(DateTime::from_timestamp(value as _, 0).unwrap().naive_utc())
+            }
+            TimeUnit::Millisecond => Timestamp(
+                DateTime::from_timestamp((value / 1_000) as _, (value % 1_000 * 1000000) as _)
+                    .unwrap()
+                    .naive_utc(),
+            ),
+            TimeUnit::Microsecond => Timestamp(
+                DateTime::from_timestamp((value / 1_000_000) as _, (value % 1_000_000 * 1000) as _)
+                    .unwrap()
+                    .naive_utc(),
+            ),
+            TimeUnit::Nanosecond => Timestamp(
+                DateTime::from_timestamp(
+                    (value / 1_000_000_000) as _,
+                    (value % 1_000_000_000) as _,
+                )
                 .unwrap()
                 .naive_utc(),
-        )
+            ),
+        }
     }
 
-    fn into_arrow(self) -> Self::ArrowType {
-        self.0
-            .signed_duration_since(NaiveDateTime::default())
-            .num_microseconds()
-            .unwrap()
+    fn into_arrow_with_unit(self, time_unit: Self::TimestampType) -> Self::ArrowType {
+        match time_unit {
+            TimeUnit::Second => self
+                .0
+                .signed_duration_since(NaiveDateTime::default())
+                .num_seconds(),
+            TimeUnit::Millisecond => self
+                .0
+                .signed_duration_since(NaiveDateTime::default())
+                .num_milliseconds(),
+            TimeUnit::Microsecond => self
+                .0
+                .signed_duration_since(NaiveDateTime::default())
+                .num_microseconds()
+                .unwrap(),
+            TimeUnit::Nanosecond => self
+                .0
+                .signed_duration_since(NaiveDateTime::default())
+                .num_nanoseconds()
+                .unwrap(),
```

Comment:
I think you can use `DateTime::timestamp_xxx()` here: https://doc.servo.org/chrono/struct.DateTime.html#method.timestamp_micros

Reply:
This is not resolved.